src/backend/metal/src/command.rs

```diff
@@ -453,176 +147,239 @@ impl command::Buffer<Resources> for CommandBuffer {
+                } else {
+                    // if no clear has been specified, we simply load the
+                    // previous content
+                    attachment.set_load_action(MTLLoadAction::Load);
+                }
+            }
+        }
 
-                unsafe {
-                    if self.encoding {
-                        self.render_encoder.end_encoding();
+        if let Some(depth) = targets.depth {
+            let attachment = render_pass_descriptor.depth_attachment();
+            attachment.set_texture(unsafe { *(depth.0) });
 
-                        if self.root {
-                            (*self.master_encoder).end_encoding();
-                            // (*self.master_encoder).release();
-                            *self.master_encoder = MTLParallelRenderCommandEncoder::nil();
-                        }
+            if let Some(layer) = depth.1 {
+                attachment.set_slice(layer as u64);
+            }
 
-                        self.encoding = false;
-                    }
-                    // self.render_encoder.release();
-                    self.render_encoder = MTLRenderCommandEncoder::nil();
-                }
+            // do we need to handle any other cases?
+            attachment.set_store_action(MTLStoreAction::Store);
+
+            let clear = unsafe { *depth.2 };
+
+            if let Some(val) = clear {
+                attachment.set_load_action(MTLLoadAction::Clear);
+                attachment.set_clear_depth(val as f64);
+
+                // same here, clear should not be persistent across frames
+                unsafe { *depth.2 = None; }
+            } else {
+                // see above
+                attachment.set_load_action(MTLLoadAction::Load);
             }
-        } else {
-            self.cache.targets = Some(targets);
+        }
+
+        self.encoder.set_render_pass_descriptor(render_pass_descriptor);
+        if let Some(dim) = targets.dimensions {
+            self.encoder.set_viewport(MTLViewport {
+                originX: 0f64,
+                originY: 0f64,
+                width: dim.0 as f64,
+                height: dim.1 as f64,
+                znear: 0f64,
+                zfar: 1f64
+            });
         }
     }
 
     fn bind_index(&mut self, buf: Buffer, idx_type: IndexType) {
-        self.index_buf = Some((buf, idx_type));
+        use map::map_index_type;
+
+        // TODO(fkaa): pass wrapper instead
+        self.encoder.set_index_buffer(unsafe { *(buf.0).0 }, map_index_type(idx_type));
     }
 
     fn set_scissor(&mut self, rect: target::Rect) {
-        self.buf.push(Command::SetScissor(MTLScissorRect {
+        // TODO(fkaa): why are getting 1x1 scissor?
+        /*self.encoder.set_scissor_rect(MTLScissorRect {
             x: rect.x as u64,
             y: rect.y as u64,
             width: (rect.x + rect.w) as u64,
             height: (rect.y + rect.h) as u64,
-        }));
+        });*/
```

Comment:
Right, so I was constantly getting a 1x1 scissor rect and I'm not sure why (default value?) So I just temporarily commented it out since nothing was using scissors from what I could tell